packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -1,49 +1,65 @@
 import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
-import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
-import { ApproveDataOptions, ButtonsOptions, ClickDataOptions, FundingType, PaypalCommerceInitializationData, PaypalCommercePaymentProcessor, PaypalCommerceScriptParams } from '../../../payment/strategies/paypal-commerce';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, RequestError } from '../../../common/error/errors';
+import { Country, CountryActionCreator, Region, UnitedStatesCodes, UNITED_STATES_CODES } from '../../../geography';
+import { OrderActionCreator } from '../../../order';
+import { PaymentActionCreator } from '../../../payment/';
+import { ApproveActions, ApproveDataOptions, ButtonsOptions, ClickDataOptions, CurrentShippingAddress, FundingType, OnCancelData, PaymentMethodInitializationData, PaypalCommerceInitializationData, PaypalCommercePaymentProcessor, PaypalCommerceScriptParams, ShippingAddress, ShippingChangeData, ShippingData } from '../../../payment/strategies/paypal-commerce';
+import { ConsignmentActionCreator } from '../../../shipping';
 import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
 import CheckoutButtonStrategy from '../checkout-button-strategy';
 
 export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
+    private _paymentMethod?: PaymentMethodInitializationData;
     private _isCredit?: boolean;
-    private _isVenmo?: boolean;
+    private _currentShippingAddress?: CurrentShippingAddress;
     private _isVenmoEnabled?: boolean;
+    private _isVenmo?: boolean;
+    private _shippingData?: ShippingData;
 
     constructor(
         private _store: CheckoutStore,
         private _checkoutActionCreator: CheckoutActionCreator,
         private _formPoster: FormPoster,
-        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
+        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor,
+        private _orderActionCreator: OrderActionCreator,
+        private _countryActionCreator: CountryActionCreator,
+        private _consignmentActionCreator: ConsignmentActionCreator,
+        private _billingAddressActionCreator: BillingAddressActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
     ) {}
 
     async initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
-        let state = this._store.getState();
-        const { initializationData } = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        const { initializationData } = this._paymentMethod;
+        const isHostedCheckoutEnabled = this._paymentMethod.initializationData?.isHostedCheckoutEnabled;
```

Comment:
It might be better to update this line with initializationData destructurization:

```
const { clientId, isHostedCheckoutEnabled, isVenmoEnabled } = initializationData || {};
```

Reply:
And right after that set `this._isVenmoEnabled = isVenmoEnabled;` to keep it closer to the initializationData variable

Reply:
fixed